### PR TITLE
Bombardier boomer

### PR DESCRIPTION
**Pull request by GuardianDll**
#### Summary
Content "New boomer zombie 'bombardier boomer'"

#### Purpose of change
Just a new zed, that has arc shape attack

#### Describe alternatives you've considered
not adding it

#### Testing
works fine, but require some balancing in range and damage, and maybe some another name/description

#### Additional context
inspired by KorGgenT
GIF 20 01 2022 19-04-03

### CODE REVIEW

File: data/json/monsters/zed_explosive.json [hunk 1]
```diff
@@ -234,5 +234,35 @@
       "PACIFIST",
       "SLUDGEPROOF"
     ]
+  },
+  {
+    "id": "mon_boomer_claymore",
+    "type": "MONSTER",
+    "name": "Claymore boomer",
+    "description": "This one looks like the default boomer, except the giant bone growth on his ribcage.",
```
Comment (chaohedgehogs) [inline data/json/monsters/zed_explosive.json:242]:
```suggestion
    "description": "Looks like a regular boomer, except for the giant bone growth on its ribcage.",
```

**Comment by Night-Pryanik** [comment]
What's the story behind this zombie? How and where did it get its explosive? What's exactly is exploding?

**Comment by GuardianDll** [comment]
Ugh
Some new blob mutaion? Idk, i just made a stuff people propose on the discord server

File: data/json/monster_special_attacks/spells.json [hunk 2]
```diff
@@ -485,5 +485,49 @@
...
+    "id": "boomer_claymore",
+    "type": "SPELL",
+    "name": { "str": "Boomer claymore attack" },
+    "description": "Nothing to see here.",
+    "valid_targets": [ "hostile", "ally", "ground" ],
+    "effect": "attack",
+    "shape": "cone",
+    "min_damage": 30,
+    "max_damage": 30,
+    "damage_type": "biological",
```
Comment (John-Candlebury) [inline data/json/monster_special_attacks/spells.json:514]: Regular explosions deal bash damage in the code.

And it would probably be smart to make the explosion deal a random amount of damage between 10-30 damage instead of just flat damage amount. You should probably consider if you want this effect to instakill fragile characters, IRC it will do so with current stats.

Comment (GuardianDll) [reply data/json/monster_special_attacks/spells.json]: Already replase it with cut damage

Comment (GuardianDll) [reply data/json/monster_special_attacks/spells.json]: Also remember its pretty rare monster, and has not so much damage against armored monsters

Comment (John-Candlebury) [reply data/json/monster_special_attacks/spells.json]: Should probably replace it with bash just for consistency sake.

Comment (GuardianDll) [reply data/json/monster_special_attacks/spells.json]: Explosion has "bullet" damage, which not exist in the spell system
Also:
IMG_20220121_061306

**Comment by Night-Pryanik** [comment]
Right now this looks like "this monster explodes because... just because". Sure, we have plenty of monsters whose abilities are justified by the same "blob magic" principle (acid zombie, boomer zombie), but in my opinion it's a clear symbol of bad design where one can justify vastly different abilities through a single reason.

**Comment by GuardianDll** [comment]
So what do you think i need to do to avid this effect?

File: data/json/monster_special_attacks/spells.json (hunk 2 above)
Comment (John-Candlebury) [reply data/json/monster_special_attacks/spells.json]: Well the explosion "blast" which always hits you and works mostly like the spell here deals bash. Shrapnel does deal bullet damage, but is also not guaranteed to always hit you.

You can actually make spells  cast normal explosions using the "explosion" spell effect. Although Im not sure if you considered this alternative already

Comment (GuardianDll) [reply data/json/monster_special_attacks/spells.json]: I never check how it works with shapes except "blast"
Ill test it and change the effect if thats would be better than current solution

**Comment by Night-Pryanik** [comment]
> So what do you think i need to do to avid this effect?

1. Invent some background for the monster which justifies existence of claymore-type explosive embedded into its body. At the moment I can't think of anything that will even remotely fit the requirement.
2. Disregard my opinion as others may be completely fine with the current situation.

**Comment by GuardianDll** [comment]
> justifies existence of claymore-type explosive embedded into its body
I dont think there is a real m18a1 on his torso, thats more about "his attack looks like work of claymore"

File: data/json/monster_special_attacks/spells.json (hunk 2 above)
Comment (GuardianDll) [reply data/json/monster_special_attacks/spells.json]: Oh btw, if you misunderstood it (as pryanik on the comment below) there is no real landmineon his ribcage, its just a dummy name

**Comment by kevingranade** [comment]
If you're looking for an immediate mechanism of action, it would probably resemble the [Bombardier Beetle](https://en.wikipedia.org/wiki/Bombardier_beetle), which uses an exothermic chemical reaction between two chemicals it secretes to generate large volumes of gas that propel nearly boiling caustic chemicals toward the target.
In the case of this boomer, the mechanical differences are volume of chemical reactants, potentially efficiency of reaction, and a lack of concern for self-preservation or metabolic efficiency.

So basically, a large proportion of the boomer's torso is full of glands that secrete chemicals and mix them on command to start the reaction, and on top of these glands are bony plates that are propelled toward the target by the resulting explosion, along with hot gasses and potentially caustic chemicals.

Bombadier Boomer might be a better name.

**Comment by AcidAntOnAMinefield** [comment]
The description has a typo: "This Zombies" (missing an apostrophe there). 

A few things I should add:
- His attack *always* hits the torso. Always. Using a test world with several different characters, not once they hit any other limb. (average 8 stats "Survivor", no relevant armour or stats/traits). 3 are enough to kill your average survivor, seeing the "always torso damage" averages to 30 points of damage and bleeding. Although I assume it's part of its "spell" ability. 

- He's too fast, for something as volatile as they are, you'd assume it wouldn't be moving as fast as it currently does, would expect something along the lines of 65-75% at best. 

These "spell" abilities are so cheap.

**Comment by GuardianDll** [comment]
> This Zombies

Damn

> His attack always hits the torso

Yeah, thats the limitation of spell system

> 3 are enough to kill your average survivor

I made them pretty rare, they're less rare then gasoline boomers, so i dont think 3 of them on the same place would be so common situation

> He's too fast

🤷

**Comment by a-chancey** [comment]
> > 3 are enough to kill your average survivor
> 
> I made them pretty rare, they're less rare then gasoline boomers, so i dont think 3 of them on the same place would be so common situation

since the attack is a spell, is it bypassing armor? If so, that would be devastating for long term survival. 

“Late game”, aka by the time first winter started, I was seeing a gasoline zombie on nearly every block in towns, multiples in mansions and any type of tower, and they’ve only gotten more prevalent now that I’m into my second spring. Half of the towns near my base are now ash piles due to damn gasoline zombies getting cut up on terrain like rose bushes.

**Comment by GuardianDll** [comment]
its not bypass the armor, it just concentrate it on your torso
got 18 damage wearing winter survivor suit, for example

**Comment by AcidAntOnAMinefield** [comment]
> So basically, a large proportion of the boomer's torso is full of glands that secrete chemicals and mix them on command to start the reaction, and on top of these glands are bony plates that are propelled toward the target by the resulting explosion, along with hot gasses and potentially caustic chemicals.

This would be a bit more acceptable if it was the case. The way it currently stands is the explosion starts on the player and may hit the boomer itself. Imo the explosion/effect should be centered on the creature itself and expand from it. If it is supposed to release "shrapnel", then the damage should be scattered (a few cuts here and there along with bleeding wounds) instead of "torso-centric" - remember the Feral Humans? Yeah. 

Heck, I'd probably even say I'd be alright if the explosion's "fallout" would be really hot hair for a few turns and stamina depleting gas. But as it stands now, it's just "cheap" imo.

**Comment by GuardianDll** [comment]
> The way it currently stands is the explosion starts on the player and may hit the boomer itself

another spell system limitation - when you make it with only 1 spell zed try to use it on the max distance instead of minimizing the distance.

> stamina depleting gas

good idea, ill do it later

File: data/json/monster_special_attacks/spells.json (hunk 2 above)
Comment (GuardianDll) [reply data/json/monster_special_attacks/spells.json]: just made some test, and of course `explosion` effect has no support for `shape`

**Comment by estebandellasilva** [comment]
... While i know its a bit late ... maybe the name can be improved on? Bombardier Boomer sounds a like it could be shortend down and combined a little

Boombardier or something along those lines. the current name iss ok but it just lacks some OOPHM and its kinda long
